hudi-common/src/main/java/org/apache/hudi/common/table/view/IncrementalTimelineSyncFileSystemView.java

```diff
@@ -108,17 +110,22 @@ private void runIncrementalSync(HoodieTimeline timeline, TimelineDiffResult diff
     LOG.info("Timeline Diff Result is :" + diffResult);
 
     // First remove pending compaction instants which were completed
-    diffResult.getFinishedCompactionInstants().stream().forEach(instant -> {
+    diffResult.getFinishedViewChangingInstants().stream().forEach(instant -> {
       try {
-        removePendingCompactionInstant(timeline, instant);
+        if (HoodieTimeline.COMPACTION_ACTION.equals(instant.getAction())) {
```

Comment:
Can we introduce something like HoodieInstant.isAction(String action) instead of directly checking the action names here ? There could be many such occurrence like this ?